Branch-TestBed/Branch-SDK Load Tests/Branch_SDK_Load_Tests.m

```diff
@@ -0,0 +1,78 @@
+//
+//  Branch_SDK_Load_Tests.m
+//  Branch-TestBed
+//
+//  Created by Qinwei Gong on 2/23/15.
+//  Copyright (c) 2015 Branch Metrics. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "Branch.h"
+#import "BNCPreferenceHelper.h"
+#import "BNCServerInterface.h"
+#import "Nocilla.h"
+
+@interface Branch_SDK_Load_Tests : XCTestCase {
+    
+@private
+    __weak Branch *branch;
+}
+
+@end
+
+@implementation Branch_SDK_Load_Tests
+
+- (void)setUp {
+    [super setUp];
+    
+    branch = [Branch getInstance:@"5668720416392049"];
+    
+    [BNCPreferenceHelper setSessionID:@"97141055400444225"];
+    [BNCPreferenceHelper setDeviceFingerprintID:@"94938498586381084"];
+    [BNCPreferenceHelper setIdentityID:@"95765863201768032"];
+    [BNCPreferenceHelper setLinkClickID:NO_STRING_VALUE];
+    [BNCPreferenceHelper setLinkClickIdentifier:NO_STRING_VALUE];
+    [BNCPreferenceHelper setSessionParams:NO_STRING_VALUE];
+    
+    [branch simulateInitFinished];
+    
+    [[LSNocilla sharedInstance] start];
+}
+
+- (void)tearDown {
+    [[LSNocilla sharedInstance] clearStubs];
+    [[LSNocilla sharedInstance] stop];
+    
+    [super tearDown];
+}
+
+- (void)testLoad {
+    NSDictionary *responseDict = @{@"url": @"https://bnc.lt/l/3PxZVFU-BK"};
+    NSData *responseData = [BNCServerInterface encodePostParams:responseDict];
+    
+    stubRequest(@"POST", [BNCPreferenceHelper getAPIURL:@"url"])
+    .andReturn(200)
+    .withHeaders(@{@"application/json": @"Content-Type"})
+    .withBody(responseData);
+    
+    XCTestExpectation *getShortURLExpectation = [self expectationWithDescription:@"Test getShortURL"];
+    
+    for (int i = 0; i < 1000; i++) {
+        [branch getShortURLWithParams:nil andChannel:[NSString stringWithFormat:@"%d", i] andFeature:nil andCallback:^(NSString *url, NSError *error) {
```

Comment:
Could you explain what this test is actually doing? It seems to be executing a 'get link' a thousand times, but I don't think that in itself is terribly valuable. The expectation is that this will return in < 1 second, but this won't work for two reasons.

* The for loop itself makes 1000 async requests, which doesn't guarantee completion before the final get url is called, so the only thing you're actually assuring there is that the last request finishes in less than one second
* Testing time is a bad idea in general because any system quirk (say the CPU on the machine is hung for a second on any arbitrary process) could cause the test to fail -- it's not a 'determinate' test.